services/artists-api/src/routes/updateProfileStatusByToken.js

```diff
@@ -0,0 +1,115 @@
+const { StatusCodes, ReasonPhrases } = require('http-status-codes');
+const logger = require('@artistdirectory/logger');
+const generateToken = require('./../utilities/generateToken');
+const emailClient = require('@artistdirectory/email-client');
+const mongodbClient = require('../models/mongodbClient');
+const models = require('../models');
+
+const { DIRECTORY_API_URL } = process.env;
+
+const handler = async (event, context) => {
+  context.callbackWaitsForEmptyEventLoop = false;
+
+  await mongodbClient.connect();
+
+  if (event.source === 'serverless-plugin-warmup') {
+    await new Promise((resolve) => setTimeout(resolve, 25));
+    return 'Lambda is warm!';
+  }
+
+  try {
+    const { reviewToken } = event.pathParameters;
+    const Artist = await models.get('Artist');
+    const { approvalStatus, rejectionReason } = JSON.parse(event.body);
+
+    // generate token for url for editing artist profile if rejected
+    const editProfileToken = await generateToken();
+    const artist = await Artist.findOneAndUpdate(
```

Comment:
The code you have here could be considered "business logic," and this kind of code should live inside the model itself. Let's move this `findOneAndUpdate()` call to the model itself. I suggest we create two methods: `artist.approveProfile()` and `artist.rejectProfile(rejectionReason)` (instance methods, not static methods). We would call one of these based on the value of `approvalStatus`.

Reply:
done